src/gui/vector/qgsattributesformproperties.cpp

```diff
@@ -1988,3 +2010,283 @@ void QgsAttributesFormProperties::updatedFields()
     }
   }
 }
+
+void QgsAttributesFormProperties::onContextMenuRequested( QPoint point )
+{
+  if ( mAvailableWidgetsTree->selectedItems().count() != 1 )
+    return;
+
+  QPoint globalPos = mAvailableWidgetsTree->viewport()->mapToGlobal( point );
+
+  QTreeWidgetItem *item = mAvailableWidgetsTree->itemAt( point );
+  const DnDTreeItemData itemData = item->data( 0, DnDTreeRole ).value<DnDTreeItemData>();
+  if ( itemData.type() == DnDTreeItemData::Field )
+  {
+    const QClipboard *clipboard = QApplication::clipboard();
+    const bool pasteEnabled = clipboard->mimeData()->hasFormat( QStringLiteral( "application/x-qgsattributetabledesignerelementclipboard" ) );
+    mActionPasteWidgetConfiguration->setEnabled( pasteEnabled );
+    mAvailableWidgetsTreeContextMenu->popup( globalPos );
+  }
+}
+
+void QgsAttributesFormProperties::copyWidgetConfiguration()
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  if ( mAvailableWidgetsTree->selectedItems().count() != 1 )
+    return;
+
+  const QTreeWidgetItem *item = mAvailableWidgetsTree->selectedItems().at( 0 );
+  const DnDTreeItemData itemData = item->data( 0, DnDTreeRole ).value<DnDTreeItemData>();
+  if ( itemData.type() != DnDTreeItemData::Field )
+    return;
+
+  const QString fieldName = item->data( 0, FieldNameRole ).toString();
+  const int index = mLayer->fields().indexOf( fieldName );
+
+  if ( index < 0 )
+    return;
+
+  const QgsField field = mLayer->fields().field( index );
+
+  // We won't copy field aliases nor comments
```

Comment:
I would discuss about aliases. I would say it's quite common to have the field in english and have the translated alias. So that would make sense to me to also copy it.

Reply:
Please post this in the original issue to discuss it with A. Poncet.

Actually, I've already used this copy&paste functionality (because it was handy for migrating a QGIS project for the Siemens project) and keeping the aliases intact seemed like a good idea. But, let's see what A. Poncet says about the use case you're mentioning.